assets/app.js

```diff
@@ -794,39 +825,54 @@ new Vue({
         },
     },
     methods: {
-        validateRejectedProduct: function (type, product) {
-            if (type == 'materials' && product['is_rejected'] && product['rejected_' + type]) {
-                if (!(product['is_rejected'] && product['rejected_' + type])) {
-                    return true
+        validateRejectedProduct: function (product) {
+            let verified_resources = {
+                'base_product': true,
+                'component1': true,
+                'component2': true,
+                'materials': true,
+                'no_deleted_materials': true
+            }
+            for (var type in verified_resources) {
+                if (type === 'no_deleted_materials') {
+                    continue
```

Comment:
???

Reply:
If you don't want to loop it, don't put it into `verified_resources` in the beginning, just add it in the end when you know if its true/false